### PR TITLE
RUM: Route transaction|span events on agent.name

### LINKED ISSUE #6223: Add a delete phase to APM ILM policies
APM data reduces in value as it ages. Users generally do not want to look at a distributed trace from 2 months ago.

We should consider adding a delete phase to APM's ILM policies to keep costs in check. Each data stream may retain data for a longer period by default; metrics consume less space and provide greater long-term value than individual traces.

### PR DESCRIPTION
**Pull request by marclop**
## Motivation/summary

When datastreams are enabled, the RUM spans and transactions are sent to
the `traces-apm.rum` datastream. The supported RUM Agent names are:
    - "js-base" (Legacy RUM agent)
    - "rum-js" (RUM agent)
    - "iOS/swift" (iOS Agent)

An ILM Delete phase has been set up per datastream as follows:

|        Data Stream     | Delete Phase Days |
| ---------------------- | ----------------- |
| `logs-apm.error`       | `10d`             |
| `metrics-apm.profiling`| `10d`             |
| `traces-apm`           | `10d`             |
| `traces-apm.rum`       | `90d`             |
| `metrics-apm.app`      | `90d`             |
| `metrics-apm.internal` | `90d`             |

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
- [x] Documentation has been updated

## How to test these changes

1. `docker-compose up -d`
2. `make`
3. `./apm-server -E apm-server.rum.enabled=true -E apm-server.data_streams.enabled=true -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme`
4. `curl -H "Content-type: application/x-ndjson" --data-binary @testdata/intake-v3/rum_events.ndjson http://localhost:8200/intake/v3/rum/events`
5. `curl -H "Content-type: application/x-ndjson" --data-binary @testdata/intake-v2/transactions_spans_rum_2.ndjson http://localhost:8200/intake/v2/rum/events`
6. Navigate to Kibana Dev Tools and issue the following query and verify: 

```javascript
GET traces-apm.rum*/_search
{
  "_source": [false], 
  "fields": [
    "processor.*",
    "data_stream.dataset",
    "agent.name"
  ], 
  "query": {
    "match_all": {}
  }
}
```
  - `data_stream.dataset == apm.rum`
  - `processor.event == transaction|span`
  - `agent.name == rum-js|js-base`

6. Verify that the transactions show up in `Observability > APM > Services`. (There should be 2 services with 1 transaction each).
7. Verify that the ILM delete phases are set up in ILM.

## Related issues

Closes #6223

### CODE REVIEW

File: apmpackage/apm/data_stream/traces.rum/elasticsearch/ingest_pipeline/apm_data_stream_migration.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (axw) [inline apmpackage/apm/data_stream/traces.rum/elasticsearch/ingest_pipeline/apm_data_stream_migration.yml:1]: We don't need to duplicate this one.

File: model/modelprocessor/datastream.go [hunk 2]
```diff
@@ -25,6 +25,13 @@ import (
 	"github.com/elastic/apm-server/model"
 )
 
+// These are all the known agents that send "RUM" data to the APM Server.
+var rumAgents = map[string]struct{}{
+	"rum-js":  {},
+	"js-base": {},
+	"iOS":     {},
```
[the remark this answers is not in the document]
Comment (marclop) [reply model/modelprocessor/datastream.go:32]: I see now that we do use `iOS/swift` as the agent name in our tests, I initially copied this value from the `apm-agent-ios` repo.

File: model/modelprocessor/datastream.go [hunk 3]
```diff
@@ -25,6 +25,13 @@ import (
...
+}
```
[the remark this answers is not in the document]
Comment (marclop) [reply model/modelprocessor/datastream.go:33]: yeah true, I initially had it as a function that checked the map but removed it, I think it makes sense to change it to a switch/case.

File: apmpackage/apm/data_stream/traces.rum/elasticsearch/ingest_pipeline/apm_data_stream_migration.yml (hunk 1 above)
Comment (marclop) [reply apmpackage/apm/data_stream/traces.rum/elasticsearch/ingest_pipeline/apm_data_stream_migration.yml:1]: copy / paste issue, thanks for spotting.

**Comment by marclop** [comment]
@bmorelli25 Do we want to document the ILM policies that are set up for the data streams anywhere more "formal"?

I see we've got the current documentation for data streams in `docs/legacy/apm-package/data-streams.asciidoc`, which I assume is going to be deleted. Where would we do so?

**Review by axw — COMMENTED** [review]
LGTM, but I'd prefer not to create a whole new section in the readme.

**Review by axw — APPROVED** [review]

**Comment by bmorelli25** [comment]
> @bmorelli25 Do we want to document the ILM policies that are set up for the data streams anywhere more "formal"?

Yes! Thanks for the ping. Opened https://github.com/elastic/observability-docs/issues/1224 to track.

**Comment by stuartnelson3** [comment]
confirmed with 50edba17e7a6e4c5cb58c6258e1a38dc2387bcdb

**Pull request merged**